### DISCUSSION TITLE
Feature request: instruct-pix2pix

### DISCUSSION THREAD

**Discussion by Jonathhhan** [leejet/stable-diffusion.cpp#61] [Ideas]
Would it be difficult to implement instruct-pix2pix, for example: https://huggingface.co/timbrooks/instruct-pix2pix ?
If I load the `ggml` model I get the error:
`stable-diffusion.cpp:3045 - tensor 'model.diffusion_model.input_blocks.0.0.weight' has wrong shape in model file: got [3, 3, 8, 320], expected [3, 3, 4, 320]`
I can run the inference, but the result is a black image.

**Comment by gogo2464** [comment]
I have exactly the same issue! @Jonathhhan Could you tell me if I can convert this file to gguf to fix the issue quick please?

I would love to do it sadly I fear I do not have the time for a PR. Sorry!

**Comment by gogo2464** [comment]
I have exactly the error:

```bash
./sd -m dir/liberty_main.instruct-pix2pix.safetensors -H 64 -W 64 --prompt "pikachu"
[INFO ] stable-diffusion.cpp:197  - loading model from 'dir/liberty_main.instruct-pix2pix.safetensors'
[INFO ] model.cpp:908  - load dir/liberty_main.instruct-pix2pix.safetensors using safetensors format
[INFO ] stable-diffusion.cpp:244  - Version: SD 1.x 
[INFO ] stable-diffusion.cpp:277  - Weight type:                 f32
[INFO ] stable-diffusion.cpp:278  - Conditioner weight type:     f32
[INFO ] stable-diffusion.cpp:279  - Diffusion model weight type: f32
[INFO ] stable-diffusion.cpp:280  - VAE weight type:             f32
  |===================>                              | 446/1131 - 0.00it/s[ERROR] model.cpp:1906 - tensor 'model.diffusion_model.input_blocks.0.0.weight' has wrong shape in model file: got [3, 3, 8, 320], expected [3, 3, 4, 320]
[WARN ] model.cpp:1791 - process tensor failed: 'model.diffusion_model.input_blocks.0.0.weight'
[ERROR] model.cpp:1922 - load tensors from file failed
[ERROR] stable-diffusion.cpp:441  - load tensors from model loader failed
new_sd_ctx_t failed
```

I may do the PR. Are you sure that this PR will fix my issue if I start it please?

**Comment by stduhpf** [comment]
I can try to give it a shot.

**Reply by stduhpf** [reply]
Seems easier than I first expected.

`sd.exe -M img2img --model instruct-pix2pix-00-22000.safetensors -p "Make it a cat" --strength 1 -i input.png --steps 100 --cfg-scale 6 --sampling-method euler_a`

| input | output |
| --- | --- | 
| ![input](https://github.com/user-attachments/assets/ed76c83c-c1fc-4759-894c-08b8b0092e14) | ![output](https://github.com/user-attachments/assets/17de93ed-e377-4dd5-9671-9bc456002fca) |

We would still need to support separate cfg scales for text and img conditionnings. for better results I think.

**Reply by stduhpf** [reply]
@Jonathhhan (I know you asked for it ages ago) and @gogo2464, you can try https://github.com/leejet/stable-diffusion.cpp/pull/679, I think it's working fine now.

**Reply by gogo2464** [reply]
@stduhpf yes it is working for me now. Thank you.
